pkg/systemd/generate/containers.go

```diff
@@ -349,6 +350,28 @@ func executeContainerTemplate(info *containerInfo, options entities.GenerateSyst
 
 		startCommand = append(startCommand, remainingCmd...)
 		startCommand = escapeSystemdArguments(startCommand)
+		if options.TemplateUnitFile {
+			info.IdentifySpecifier = true
+			// in order not to override different services with the same container
+			// we have to change the "--name" operator to contain the %i specifier so
+			// it will be unique to the service. However, if the start command does not contain
+			// a name we will return an error message.
```

Comment:
I understand that we need to change the `--name` but I don't understand why `--name` _must_ be set.  There are no conflicts if there's no `--name` flag ... Podman will take care of assigning a conflict-free name.